torch/csrc/distributed/rpc/py_rref.cpp

```diff
@@ -46,6 +47,46 @@ RRefForkData fromPyTuple(const py::tuple& pyTuple) {
 
   return RRefForkData(ownerId, rrefId, forkId, parent, typeStr);
 }
+
+c10::optional<TypePtr> typePyObjToTypePtr(const py::object& type_value) {
+  if (type_value.is(py::none())) {
+    return c10::nullopt;
+  }
+  c10::QualifiedName type_value_str = c10::QualifiedName(py::cast<std::string>(
+      py::module::import("torch.jit").attr("_qualified_name")(type_value)));
```

Comment:
for non-existing type value, will this function return an empty option or throw an error?

Reply:
I think it should throw since it's a user provided type_hint, which should be mapped to a known JIT type, let me update.